app/modules/asset_groups/models.py

```diff
@@ -286,9 +286,37 @@ def get_completion(self):
         # calculation generates a floating point value, reporting that would be claiming precision without accuracy
         return round(completion)
 
+    # returns a getter for a given config field, allowing for casting and default vals
+    def config_field_getter(field_name, default=None, cast=None):
```

Comment:
I didn't realize this last time but actually, this is not a normal method because it doesn't take "self" as the first argument.  When I was writing the test, I did `ags.config_field_getter("field_name")` and that was wrong because what actually happened was `config_field_getter(field_name=ags, default="field_name")`.  To avoid this confusion, I suggest adding `@staticmethod` before this method so even if you do `ags.config_field_getter("field_name")`, the arguments will be correct.